tests/e2e/cucumber/step_definitions/app.files.ts

```diff
@@ -99,6 +99,46 @@ When(
   }
 )
 
+When(
+  /^"([^"]*)" creates a public link to the following resource(s)? via the (sidebar panel|quick action)$/,
+  async function (
+    this: World,
+    stepUser: string,
+    _: string,
+    actionType: string,
+    stepTable: DataTable
+  ) {
+    const actor = this.actorsEnvironment.getActor({ id: stepUser })
+    const { publicLink: publicLinkPage } = new FilesPage({ actor })
+    const shareInfo = stepTable.hashes()
+    for (const linkShare of shareInfo) {
+      const { resource, name, role, dateOfExpiration, password } = linkShare
+      await publicLinkPage.createPublicLinkForResource({
+        resource,
+        name,
+        role,
+        dateOfExpiration,
+        password,
+        via: actionType === 'quick action' ? 'QUICK_ACTION' : 'SIDEBAR_PANEL'
+      })
+    }
+  }
+)
+
+Then(
+  '{string} should see {int} public link',
+  async function (this: World, stepUser: string, noOfPublicLink: number): Promise<void> {
+    const actor = this.actorsEnvironment.getActor({ id: stepUser })
+    const { publicLink: publicLinkPage } = new FilesPage({ actor })
+    const totalNoPublicLinks = await publicLinkPage.getLinkCount()
+    if (totalNoPublicLinks !== noOfPublicLink) {
+      throw new Error(
+        'Expected no of link to be created is ' + noOfPublicLink + ' but got ' + totalNoPublicLinks
+      )
+    }
```

Comment:
please, use `assert`. It makes sense in the test although the internal workings are similar